### PR TITLE
Make MAX_IDLE_THREADS configurable for libfuse3

### PR DESCRIPTION
**Pull request by ssz1997**
Set MAX_IDLE_THREADS to a default value of 64 for libfuse3

### CODE REVIEW

**Comment by ssz1997** [comment]
@LuQQiu PTAL. Thx!

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 1]
```diff
@@ -7342,6 +7350,10 @@ public static final class Name {
     public static final String FUSE_JNIFUSE_ENABLED = "alluxio.fuse.jnifuse.enabled";
     public static final String FUSE_SHARED_CACHING_READER_ENABLED
         = "alluxio.fuse.shared.caching.reader.enabled";
+    public static final String FUSE_JNIFUSE_LIBFUSE_VERSION =
```
Comment (ssz1997) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:7353]: Originally the libfuse version property key was under section "Standalone FUSE process related properties", which is not correct.

File: integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java [hunk 2]
```diff
@@ -72,10 +73,11 @@ public void mount(boolean blocking, boolean debug, String[] fuseOpts) {
     if (mountPointStr.endsWith("\\")) {
       mountPointStr = mountPointStr.substring(0, mountPointStr.length() - 1);
     }
+    String maxIdleThreadsStr = "max_idle_threads=" + maxIdleThreads;
     if (!debug) {
-      arg = new String[] {getFileSystemName(), "-f", mountPointStr};
+      arg = new String[] {getFileSystemName(), "-f", mountPointStr, maxIdleThreadsStr};
     } else {
-      arg = new String[] {getFileSystemName(), "-f", "-d", mountPointStr};
+      arg = new String[] {getFileSystemName(), "-f", "-d", mountPointStr, maxIdleThreadsStr};
```
Comment (LuQQiu) [inline integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java:80]: @ssz1997 no need to change on the JNIFuse side?

Comment (ssz1997) [reply integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java:80]: From my testing on linux there's no need to change JNIFuse, and I can see more than 10 threads in WAITING or TIMED_WAITING stage when I run stressbench.

File: core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 3]
```diff
@@ -5624,6 +5624,14 @@ public String toString() {
               + "other value to use libfuse2 first, libfuse3 if libfuse2 failed")
           .setScope(Scope.ALL)
           .build();
+  public static final PropertyKey FUSE_LIBFUSE3_MAX_IDLE_THREADS =
```
Comment (LuQQiu) [inline core/common/src/main/java/alluxio/conf/PropertyKey.java:5627]: We may need a consolidated story about how to set related Fuse configurations.
@ssz1997 can you help understand how many different kinds of config Fuse exposes and see can we have a consolidated way of dealing with each kind of config?

e.g. Fuse mount options
e.g. other Fuse init parameters like MAX_IDLE_THREADS
....
may be others.

Thanks for supporting MAX_IDLE_THREADS. I am a little worried about we keep adding new config like this and have multiple ways of passing fuse config in (especially directly pass the config along the whole path)

File: integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java [hunk 4]
```diff
@@ -59,9 +59,10 @@ public AbstractFuseFileSystem(Path mountPoint) {
    *
    * @param blocking whether this command is blocking
    * @param debug whether to show debug information
+   * @param maxIdleThreads maximum number of idle threads enforced by libfuse3
    * @param fuseOpts
    */
-  public void mount(boolean blocking, boolean debug, String[] fuseOpts) {
+  public void mount(boolean blocking, boolean debug, int maxIdleThreads, String[] fuseOpts) {
```
Comment (LuQQiu) [inline integration/jnifuse/fs/src/main/java/alluxio/jnifuse/AbstractFuseFileSystem.java:65]: I would expect there is a MountConfiguration class holding all those information, and the whole class object can be formed when launches Fuse and all intermediate steps just directly pass this object or take information from it (without modifying)

**Comment by ssz1997** [comment]
Libfuse3 investigation is paused. Close this PR

**Pull request closed without merging**